haystack_experimental/dataclasses/chat_message.py

```diff
@@ -67,6 +69,98 @@ def __repr__(self) -> str:
+            when working with models that have resolution constraints or when transmitting images to remote services.
+        :param mime_type:
+            Optional MIME type of the image. If not provided, the MIME type is guessed from the file extension.
+        :param detail:
+            Optional detail level of the image (only supported by OpenAI). One of "auto", "high", or "low".
+        :param meta:
+            Additional metadata for the image.
+
+        :returns:
+            An ImageContent object.
+        """
+        # to avoid a circular import
+        from haystack_experimental.components.image_converters import ImageFileToImageContent
+
+        if mime_type:
+            source = ByteStream.from_file_path(Path(file_path), mime_type=mime_type, meta={"file_path": str(file_path)})
+        else:
+            source = file_path
+
+        converter = ImageFileToImageContent(size=size, detail=detail)
+        result = converter.run(sources=[source], meta=[meta] if meta else None)
+        return result["image_contents"][0]
+
+
+    @classmethod
+    def from_url(
+        cls,
+        url: str,
+        *,
+        retry_attempts: int = 2,
+        timeout: int = 10,
+        size: Optional[Tuple[int, int]] = None,
+        mime_type: Optional[str] = None,
+        detail: Optional[Literal["auto", "high", "low"]] = None,
+        meta: Optional[Dict[str, Any]] = None,
+    ) -> "ImageContent":
+        """
+        Create an ImageContent object from a URL.
+
+        :param url:
+            The URL of the image.
+        :param retry_attempts:
+            The number of times to retry to fetch the URL's content.
+        :param timeout:
+            Timeout in seconds for the request.
+        :param size:
+            If provided, resizes the image to fit within the specified dimensions (width, height) while
+            maintaining aspect ratio. This reduces file size, memory usage, and processing time, which is beneficial
+            when working with models that have resolution constraints or when transmitting images to remote services.
+        :param mime_type:
+            Optional MIME type of the image. If not provided, the MIME type is guessed from the content type of the URL.
+        :param detail:
+            Optional detail level of the image (only supported by OpenAI). One of "auto", "high", or "low".
+        :param meta:
+            Additional metadata for the image.
+
+        :returns:
+            An ImageContent object.
+        """
+        # to avoid a circular import
+        from haystack_experimental.components.image_converters import ImageFileToImageContent
+
+        fetcher = LinkContentFetcher(raise_on_failure=True, retry_attempts=retry_attempts, timeout=timeout)
+        bytestream = fetcher.run(urls=[url])["streams"][0]
+
+        if mime_type:
+            bytestream.mime_type = mime_type
```

Comment:
I'm thinking we should do some additional checking here. Basically from what I can see we aren't warning the user if they provide a URL that doesn't actually point to an image. E.g. if they point to a normal website or a PDF link then there will probably a be a hard to debug error that occurs in the ImageFileToImageContent component. 

So I wonder if we should print a warning message if the inferred mime type from the LinkContentFetcher isn't an expected image format. What do you think?

Reply:
I started to work on this suggestion, then I realized that the code and the behavior of the method become complex if we still want to allow the user to provide a MIME type, so I would be inclined to remove the `mime_type` parameters from these methods. As discussed in https://github.com/deepset-ai/haystack-experimental/pull/294#issue-3042393043, given a file path or URL, MIME type inference is generally reliable.
Do you agree with this idea?

Reply:
> Do you agree with this idea?

Yes! That sounds good to me.

Reply:
- I removed the `mime_type` parameter.
- I added a check for valid image MIME types. If the MIME type does not point to an image, we now raise an error.